app/modules/wallet/wallet.controller.ts

```diff
@@ -0,0 +1,55 @@
+import * as restify from 'restify';
+import { InternalServerError, NotFoundError } from 'restify-errors';
+import IController from '../../interfaces/utils/IController';
+import IRequest from '../../interfaces/utils/IServer';
+import Wallet from '../../models/wallet.model';
+import { IWallet } from '../../interfaces/models';
+
+export default class WalletsController implements IController {
+  public async post(req: IRequest, res: restify.Response, next: restify.Next) {
+    try {
+      const walletObj = req.body;
+      const wallet = <IWallet> await Wallet.create({
+        client_id: walletObj.client_id,
```

Comment:
client_id is not coming from the body, its in the request object. `req.client_id` is the current client's id which is using these credentials to make the request. If the object is same, maybe just pass it in create function.
```javascript
Wallet.create({
  ...walletObj,
  client_id: req.client_id
});
```